OutOfMemoryError running backend Integration Tests

<!--
- Please follow the issue template below for bug reports.
- If you have a support request rather than a bug, please use [Stack Overflow](http://stackoverflow.com/questions/tagged/jhipster) with the JHipster tag.
- For bug reports it is mandatory to run the command `jhipster info` in your project's root folder, and paste the result here.
- Tickets opened without any of these pieces of information will be **closed** without any explanation.
-
- If this is a potential security issue, please read our security policy and contact us privately first: https://github.com/jhipster/generator-jhipster/security/policy
-->

##### **Overview of the issue**

<!-- Explain the bug, if an error is being thrown a stack trace helps -->

If there are several entities, the integration tests are failing with an `OutOfMemoryError`.

```log
Abc7ResourceIT > putNewAbc7() FAILED
    java.lang.IllegalStateException at DefaultCacheAwareContextLoaderDelegate.java:132
        Caused by: org.springframework.beans.factory.BeanCreationException at AbstractAutowireCapableBeanFactory.java:628
            Caused by: org.springframework.beans.factory.UnsatisfiedDependencyException at ConstructorResolver.java:800
                Caused by: org.springframework.beans.factory.BeanCreationException at ConstructorResolver.java:658
                    Caused by: org.springframework.beans.BeanInstantiationException at SimpleInstantiationStrategy.java:185
                        Caused by: org.springframework.beans.factory.UnsatisfiedDependencyException at ConstructorResolver.java:800
                            Caused by: org.springframework.beans.factory.BeanCreationException at AbstractAutowireCapableBeanFactory.java:628    
                                Caused by: java.lang.OutOfMemoryError
```

The tests are failing mostly around the 15th entity.

This is also happening in CI. Reproduced in Azure DevOps and GitHub.

And the integration tests are **very slow** compared to before.

##### **Motivation for or Use Case**

<!-- Explain why this is a bug for you -->

Tests should not fail with OutOfMemory error.

##### **Reproduce the error**

<!-- For bug reports, an unambiguous set of steps to reproduce the error -->

1. Generate ~30 Entities
2. Run backend integration tests: `./gradlew integrationTest -x webapp -x webapp_test`

##### **Related issues**

<!-- Has a similar issue been reported before? Please search both closed & open issues -->

##### **Suggest a Fix**

<!-- If you can't fix the bug yourself, perhaps you can point to what might be
  causing the problem (line of code or commit) -->

Something changed between 7.6.0 and 7.8.1, which causes this error. I'm currently using version 7.6.0 and wanted to Upgrade to the newest version. Earlier I tried to upgrade to 7.7.0 and as far as I recall the problem was not there (but I'm not 100 % sure). 

##### **JHipster Version(s)**

<!--
Which version of JHipster are you using, is it a regression?
-->

7.8.1

##### **JHipster configuration**

<!--
To provide all information we need, you should run `jhipster info` in the project root folder, and
copy/paste the result here.
The `.yo-rc.json` file generated in the root folder is mandatory for bug reports. This will help us to replicate the scenario.
You should remove any sensitive information like the rememberMe key or the jwtSecretKey key.
-->

  <details>
  <summary>JDL definitions</summary>
  <pre>
    application {
        config {
            baseName sample
            packageName sample
            applicationType monolith
            authenticationType oauth2
            buildTool gradle
            cacheProvider ehcache
            enableHibernateCache true
            clientFramework angularX
            clientPackageManager npm
            clientTheme none
            databaseType sql
            prodDatabaseType postgresql
            devDatabaseType postgresql
            messageBroker kafka
            languages [de, en]
            nativeLanguage de
            testFrameworks [cypress]
            websocket spring-websocket
        }
        entities *
    }

    entity Abc0 {
        name String unique required
        otherField String
    }

    entity Abc1 {
        name String unique required
        otherField String
    }

    entity Abc2 {
        name String unique required
        otherField String
    }

    entity Abc3 {
        name String unique required
        otherField String
    }

    entity Abc4 {
        name String unique required
        otherField String
    }

    entity Abc5 {
        name String unique required
        otherField String
    }

    entity Abc6 {
        name String unique required
        otherField String
    }

    entity Abc7 {
        name String unique required
        otherField String
    }

    entity Abc8 {
        name String unique required
        otherField String
    }

    entity Abc9 {
        name String unique required
        otherField String
    }

    entity Abc10 {
        name String unique required
        otherField String
    }

    entity Abc11 {
        name String unique required
        otherField String
    }

    entity Abc12 {
        name String unique required
        otherField String
    }

    entity Abc13 {
        name String unique required
        otherField String
    }

    entity Abc14 {
        name String unique required
        otherField String
    }

    entity Abc15 {
        name String unique required
        otherField String
    }

    entity Abc16 {
        name String unique required
        otherField String
    }

    entity Abc17 {
        name String unique required
        otherField String
    }

    entity Abc18 {
        name String unique required
        otherField String
    }

    entity Abc19 {
        name String unique required
        otherField String
    }

    entity Abc20 {
        name String unique required
        otherField String
    }

    entity Abc21 {
        name String unique required
        otherField String
    }

    entity Abc22 {
        name String unique required
        otherField String
    }

    entity Abc23 {
        name String unique required
        otherField String
    }

    entity Abc24 {
        name String unique required
        otherField String
    }

    entity Abc25 {
        name String unique required
        otherField String
    }

    entity Abc26 {
        name String unique required
        otherField String
    }

    entity Abc27 {
        name String unique required
        otherField String
    }

    entity Abc28 {
        name String unique required
        otherField String
    }

    entity Abc29 {
        name String unique required
        otherField String
    }
  </pre>
  </details>

##### **Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

See [my sample app](https://github.com/egvimo/jhipster-sample-app) for full example.

<!--
If the error is during an entity creation or associated with a specific entity.
If you are using JDL, please share that configuration as well.
-->

##### **Browsers and Operating System**

<!-- What OS are you on? is this a problem with all browsers or only IE8? -->

- Windows 10
- Azure DevOps CI
- GitHub Actions

- [x] Checking this box is mandatory (this is just to show you read everything)

<!-- Love JHipster? Please consider supporting our collective:
👉  https://opencollective.com/generator-jhipster/donate -->

--- a/generators/server/templates/src/test/java/package/config/TestContainersSpringContextCustomizerFactory.java.ejs
+++ b/generators/server/templates/src/test/java/package/config/TestContainersSpringContextCustomizerFactory.java.ejs
@@ -50,6 +50,8 @@ public class TestContainersSpringContextCustomizerFactory implements ContextCust
 
     private Logger log = LoggerFactory.getLogger(TestContainersSpringContextCustomizerFactory.class);
 
+    private static ContextCustomizer contextCustomizer;
+
 <%_ if (cacheProviderRedis) { _%>
     private static RedisTestContainer redisBean;
 <%_ } _%>
@@ -81,25 +83,26 @@ public class TestContainersSpringContextCustomizerFactory implements ContextCust
     @Override
     public ContextCustomizer createContextCustomizer(Class<?> testClass, List<ContextConfigurationAttributes> configAttributes) {
 
-        return (context, mergedConfig) -> {
-            ConfigurableListableBeanFactory beanFactory = context.getBeanFactory();
-            TestPropertyValues testValues = TestPropertyValues.empty();
+        if (null == contextCustomizer) {
+            contextCustomizer = (context, mergedConfig) -> {
+                ConfigurableListableBeanFactory beanFactory = context.getBeanFactory();
+                TestPropertyValues testValues = TestPropertyValues.empty();
 <%_ if (databaseTypeCouchbase || searchEngineCouchbase) { _%>
-            EmbeddedCouchbase couchbaseAnnotation = AnnotatedElementUtils.findMergedAnnotation(testClass, EmbeddedCouchbase.class);
-            if (null != couchbaseAnnotation) {
-                log.debug("detected the EmbeddedCouchbase annotation on class {}", testClass.getName());
-                log.info("Warming up the Couchbase database");
-                if (null == couchbaseBean) {
-                    couchbaseBean = beanFactory.createBean(CouchbaseTestContainer.class);
-                    beanFactory.registerSingleton(CouchbaseTestContainer.class.getName(), couchbaseBean);
-                    // ((DefaultListableBeanFactory)beanFactory).registerDisposableBean(CouchbaseTestContainer.class.getName(), couchbaseBean);
+                EmbeddedCouchbase couchbaseAnnotation = AnnotatedElementUtils.findMergedAnnotation(testClass, EmbeddedCouchbase.class);
+                if (null != couchbaseAnnotation) {
+                    log.debug("detected the EmbeddedCouchbase annotation on class {}", testClass.getName());
+                    log.info("Warming up the Couchbase database");
+                    if (null == couchbaseBean) {
+                        couchbaseBean = beanFactory.createBean(CouchbaseTestContainer.class);
+                        beanFactory.registerSingleton(CouchbaseTestContainer.class.getName(), couchbaseBean);
+                        // ((DefaultListableBeanFactory)beanFactory).registerDisposableBean(CouchbaseTestContainer.class.getName(), couchbaseBean);
+                    }
+                    testValues = testValues.and("spring.couchbase.connection-string=" + couchbaseBean.getCouchbaseContainer().getConnectionString());
+                    testValues = testValues.and("spring.couchbase.username=" + couchbaseBean.getCouchbaseContainer().getUsername());
+                    testValues = testValues.and("spring.couchbase.password=" + couchbaseBean.getCouchbaseContainer().getPassword());
+                    testValues = testValues.and("jhipster.database.couchbase.bucket-name=" + couchbaseBean.getBucketName());
+                    testValues = testValues.and("jhipster.database.couchbase.scope-name=testScope");
                 }
-                testValues = testValues.and("spring.couchbase.connection-string=" + couchbaseBean.getCouchbaseContainer().getConnectionString());
-                testValues = testValues.and("spring.couchbase.username=" + couchbaseBean.getCouchbaseContainer().getUsername());
-                testValues = testValues.and("spring.couchbase.password=" + couchbaseBean.getCouchbaseContainer().getPassword());
-                testValues = testValues.and("jhipster.database.couchbase.bucket-name=" + couchbaseBean.getBucketName());
-                testValues = testValues.and("jhipster.database.couchbase.scope-name=testScope");
-            }
 <%_ } _%>
 <%_ if (databaseTypeMongodb) { _%>
                 EmbeddedMongo mongoAnnotation = AnnotatedElementUtils.findMergedAnnotation(testClass, EmbeddedMongo.class);
@@ -128,123 +131,125 @@ public class TestContainersSpringContextCustomizerFactory implements ContextCust
                 }
 <%_ } _%>
 <%_ if (databaseTypeSql) { _%>
-            EmbeddedSQL sqlAnnotation = AnnotatedElementUtils.findMergedAnnotation(testClass, EmbeddedSQL.class);
-            if (null != sqlAnnotation) {
-                log.debug("detected the EmbeddedSQL annotation on class {}", testClass.getName());
-                log.info("Warming up the sql database");
+                EmbeddedSQL sqlAnnotation = AnnotatedElementUtils.findMergedAnnotation(testClass, EmbeddedSQL.class);
+                if (null != sqlAnnotation) {
+                    log.debug("detected the EmbeddedSQL annotation on class {}", testClass.getName());
+                    log.info("Warming up the sql database");
   <%_ if (devDatabaseTypeMysql || devDatabaseTypeMariadb || devDatabaseTypeMssql || devDatabaseTypePostgres) { _%>
-                if (Arrays.asList(context.getEnvironment().getActiveProfiles()).contains("test" + JHipsterConstants.SPRING_PROFILE_DEVELOPMENT)) {
-                    if (null == devTestContainer) {
-                        try {
-                            Class<? extends SqlTestContainer> containerClass = (Class<? extends SqlTestContainer>) Class.forName(this.getClass().getPackageName() + ".<% if (devDatabaseTypeMysql) { %>Mysql<% } else if (devDatabaseTypeMariadb) { %>Mariadb<% } else if (devDatabaseTypeMssql) { %>MsSql<% } else if (devDatabaseTypePostgres) { %>PostgreSql<% } %>TestContainer");
-                            devTestContainer = beanFactory.createBean(containerClass);
-                            beanFactory.registerSingleton(containerClass.getName(), devTestContainer);
-                            // ((DefaultListableBeanFactory)beanFactory).registerDisposableBean(containerClass.getName(), devTestContainer);
-                        } catch (ClassNotFoundException e) {
-                            throw new RuntimeException(e);
+                    if (Arrays.asList(context.getEnvironment().getActiveProfiles()).contains("test" + JHipsterConstants.SPRING_PROFILE_DEVELOPMENT)) {
+                        if (null == devTestContainer) {
+                            try {
+                                Class<? extends SqlTestContainer> containerClass = (Class<? extends SqlTestContainer>) Class.forName(this.getClass().getPackageName() + ".<% if (devDatabaseTypeMysql) { %>Mysql<% } else if (devDatabaseTypeMariadb) { %>Mariadb<% } else if (devDatabaseTypeMssql) { %>MsSql<% } else if (devDatabaseTypePostgres) { %>PostgreSql<% } %>TestContainer");
+                                devTestContainer = beanFactory.createBean(containerClass);
+                                beanFactory.registerSingleton(containerClass.getName(), devTestContainer);
+                                // ((DefaultListableBeanFactory)beanFactory).registerDisposableBean(containerClass.getName(), devTestContainer);
+                            } catch (ClassNotFoundException e) {
+                                throw new RuntimeException(e);
+                            }
                         }
-                    }
     <%_ if (reactive) { _%>
-                    testValues = testValues.and("spring.r2dbc.url=" + devTestContainer.getTestContainer().getJdbcUrl().replace("jdbc", "r2dbc")<% if (devDatabaseTypeMysql) { %>.replace("mysql", "mariadb")<% } else if (devDatabaseTypeMssql) { %>.replace(";encrypt=false", "")<% } %> + "<%- this.getDBCExtraOption(devDatabaseType) %>");
-                    testValues = testValues.and("spring.r2dbc.username=" + devTestContainer.getTestContainer().getUsername());
-                    testValues = testValues.and("spring.r2dbc.password=" + devTestContainer.getTestContainer().getPassword());
-                    testValues = testValues.and("spring.liquibase.url=" + devTestContainer.getTestContainer().getJdbcUrl() + "<%- this.getDBCExtraOption(devDatabaseType) %>" );
+                        testValues = testValues.and("spring.r2dbc.url=" + devTestContainer.getTestContainer().getJdbcUrl().replace("jdbc", "r2dbc")<% if (devDatabaseTypeMysql) { %>.replace("mysql", "mariadb")<% } else if (devDatabaseTypeMssql) { %>.replace(";encrypt=false", "")<% } %> + "<%- this.getDBCExtraOption(devDatabaseType) %>");
+                        testValues = testValues.and("spring.r2dbc.username=" + devTestContainer.getTestContainer().getUsername());
+                        testValues = testValues.and("spring.r2dbc.password=" + devTestContainer.getTestContainer().getPassword());
+                        testValues = testValues.and("spring.liquibase.url=" + devTestContainer.getTestContainer().getJdbcUrl() + "<%- this.getDBCExtraOption(devDatabaseType) %>" );
     <%_ } else { _%>
-                    testValues = testValues.and("spring.datasource.url=" + devTestContainer.getTestContainer().getJdbcUrl() + "<%- this.getDBCExtraOption(devDatabaseType) %>");
-                    testValues = testValues.and("spring.datasource.username=" + devTestContainer.getTestContainer().getUsername());
-                    testValues = testValues.and("spring.datasource.password=" + devTestContainer.getTestContainer().getPassword());
+                        testValues = testValues.and("spring.datasource.url=" + devTestContainer.getTestContainer().getJdbcUrl() + "<%- this.getDBCExtraOption(devDatabaseType) %>");
+                        testValues = testValues.and("spring.datasource.username=" + devTestContainer.getTestContainer().getUsername());
+                        testValues = testValues.and("spring.datasource.password=" + devTestContainer.getTestContainer().getPassword());
     <%_ } _%>
-                }
+                    }
   <%_ } _%>
   <%_ if (prodDatabaseTypeMysql || prodDatabaseTypeMariadb || prodDatabaseTypeMssql || prodDatabaseTypePostgres) { _%>
-                if (Arrays.asList(context.getEnvironment().getActiveProfiles()).contains("test" + JHipsterConstants.SPRING_PROFILE_PRODUCTION)) {
-                    if (null == prodTestContainer) {
-                        try {
-                            Class<? extends SqlTestContainer> containerClass = (Class<? extends SqlTestContainer>) Class.forName(this.getClass().getPackageName() + ".<% if (prodDatabaseTypeMysql) { %>Mysql<% } else if (prodDatabaseTypeMariadb) { %>Mariadb<% } else if (prodDatabaseTypeMssql) { %>MsSql<% } else if (prodDatabaseTypePostgres) { %>PostgreSql<% } %>TestContainer");
-                            prodTestContainer = beanFactory.createBean(containerClass);
-                            beanFactory.registerSingleton(containerClass.getName(), prodTestContainer);
-                            // ((DefaultListableBeanFactory)beanFactory).registerDisposableBean(containerClass.getName(), prodTestContainer);
-                        } catch (ClassNotFoundException e) {
-                            throw new RuntimeException(e);
+                    if (Arrays.asList(context.getEnvironment().getActiveProfiles()).contains("test" + JHipsterConstants.SPRING_PROFILE_PRODUCTION)) {
+                        if (null == prodTestContainer) {
+                            try {
+                                Class<? extends SqlTestContainer> containerClass = (Class<? extends SqlTestContainer>) Class.forName(this.getClass().getPackageName() + ".<% if (prodDatabaseTypeMysql) { %>Mysql<% } else if (prodDatabaseTypeMariadb) { %>Mariadb<% } else if (prodDatabaseTypeMssql) { %>MsSql<% } else if (prodDatabaseTypePostgres) { %>PostgreSql<% } %>TestContainer");
+                                prodTestContainer = beanFactory.createBean(containerClass);
+                                beanFactory.registerSingleton(containerClass.getName(), prodTestContainer);
+                                // ((DefaultListableBeanFactory)beanFactory).registerDisposableBean(containerClass.getName(), prodTestContainer);
+                            } catch (ClassNotFoundException e) {
+                                throw new RuntimeException(e);
+                            }
                         }
-                    }
     <%_ if (reactive) { _%>
-                    testValues = testValues.and("spring.r2dbc.url=" + prodTestContainer.getTestContainer().getJdbcUrl().replace("jdbc", "r2dbc")<% if (prodDatabaseTypeMysql) { %>.replace("mysql", "mariadb")<% } else if (prodDatabaseTypeMssql) { %>.replace(";encrypt=false", "")<% } %> + "<%- this.getDBCExtraOption(prodDatabaseType) %>");
-                    testValues = testValues.and("spring.r2dbc.username=" + prodTestContainer.getTestContainer().getUsername());
-                    testValues = testValues.and("spring.r2dbc.password=" + prodTestContainer.getTestContainer().getPassword());
-                    testValues = testValues.and("spring.liquibase.url=" + prodTestContainer.getTestContainer().getJdbcUrl()  + "<%- this.getDBCExtraOption(prodDatabaseType) %>");
+                        testValues = testValues.and("spring.r2dbc.url=" + prodTestContainer.getTestContainer().getJdbcUrl().replace("jdbc", "r2dbc")<% if (prodDatabaseTypeMysql) { %>.replace("mysql", "mariadb")<% } else if (prodDatabaseTypeMssql) { %>.replace(";encrypt=false", "")<% } %> + "<%- this.getDBCExtraOption(prodDatabaseType) %>");
+                        testValues = testValues.and("spring.r2dbc.username=" + prodTestContainer.getTestContainer().getUsername());
+                        testValues = testValues.and("spring.r2dbc.password=" + prodTestContainer.getTestContainer().getPassword());
+                        testValues = testValues.and("spring.liquibase.url=" + prodTestContainer.getTestContainer().getJdbcUrl()  + "<%- this.getDBCExtraOption(prodDatabaseType) %>");
     <%_ } else { _%>
-                    testValues = testValues.and("spring.datasource.url=" + prodTestContainer.getTestContainer().getJdbcUrl() + "<%- this.getDBCExtraOption(prodDatabaseType) %>");
-                    testValues = testValues.and("spring.datasource.username=" + prodTestContainer.getTestContainer().getUsername());
-                    testValues = testValues.and("spring.datasource.password=" + prodTestContainer.getTestContainer().getPassword());
+                        testValues = testValues.and("spring.datasource.url=" + prodTestContainer.getTestContainer().getJdbcUrl() + "<%- this.getDBCExtraOption(prodDatabaseType) %>");
+                        testValues = testValues.and("spring.datasource.username=" + prodTestContainer.getTestContainer().getUsername());
+                        testValues = testValues.and("spring.datasource.password=" + prodTestContainer.getTestContainer().getPassword());
     <%_ } _%>
-                }
+                    }
   <%_ } _%>
-            }
+                }
 <%_ } _%>
 <%_ if (databaseTypeCassandra) { _%>
-            EmbeddedCassandra cassandraAnnotation = AnnotatedElementUtils.findMergedAnnotation(testClass, EmbeddedCassandra.class);
-            if (null != cassandraAnnotation) {
-                log.debug("detected the EmbeddedCassandra annotation on class {}", testClass.getName());
-                log.info("Warming up the cassandra database");
-                if (null == cassandraBean) {
-                    cassandraBean = beanFactory.createBean(CassandraTestContainer.class);
-                    beanFactory.registerSingleton(CassandraTestContainer.class.getName(), cassandraBean);
-                    // ((DefaultListableBeanFactory)beanFactory).registerDisposableBean(CassandraTestContainer.class.getName(), cassandraBean);
+                EmbeddedCassandra cassandraAnnotation = AnnotatedElementUtils.findMergedAnnotation(testClass, EmbeddedCassandra.class);
+                if (null != cassandraAnnotation) {
+                    log.debug("detected the EmbeddedCassandra annotation on class {}", testClass.getName());
+                    log.info("Warming up the cassandra database");
+                    if (null == cassandraBean) {
+                        cassandraBean = beanFactory.createBean(CassandraTestContainer.class);
+                        beanFactory.registerSingleton(CassandraTestContainer.class.getName(), cassandraBean);
+                        // ((DefaultListableBeanFactory)beanFactory).registerDisposableBean(CassandraTestContainer.class.getName(), cassandraBean);
+                    }
+                    testValues = testValues.and("spring.data.cassandra.port=" +  cassandraBean.getCassandraContainer().getMappedPort(CassandraContainer.CQL_PORT))
+                        .and("spring.data.cassandra.contact-points=" +  cassandraBean.getCassandraContainer().getHost())
+                        .and("spring.data.cassandra.keyspace-name=" + CQLDataLoader.DEFAULT_KEYSPACE_NAME)
+                        .and("spring.data.cassandra.local-datacenter=" + cassandraBean.getCassandraContainer().getCluster().getMetadata().getAllHosts().iterator().next().getDatacenter())
+                        .and("spring.data.cassandra.cluster-name=" + cassandraBean.getCassandraContainer().getCluster().getMetadata().getClusterName());
                 }
-                testValues = testValues.and("spring.data.cassandra.port=" +  cassandraBean.getCassandraContainer().getMappedPort(CassandraContainer.CQL_PORT))
-                    .and("spring.data.cassandra.contact-points=" +  cassandraBean.getCassandraContainer().getHost())
-                    .and("spring.data.cassandra.keyspace-name=" + CQLDataLoader.DEFAULT_KEYSPACE_NAME)
-                    .and("spring.data.cassandra.local-datacenter=" + cassandraBean.getCassandraContainer().getCluster().getMetadata().getAllHosts().iterator().next().getDatacenter())
-                    .and("spring.data.cassandra.cluster-name=" + cassandraBean.getCassandraContainer().getCluster().getMetadata().getClusterName());
-            }
 <%_ } _%>
 <%_ if (databaseTypeNeo4j) { _%>
-            EmbeddedNeo4j neo4jAnnotation = AnnotatedElementUtils.findMergedAnnotation(testClass, EmbeddedNeo4j.class);
-            if (null != neo4jAnnotation) {
-                log.debug("detected the EmbeddedNeo4j annotation on class {}", testClass.getName());
-                log.info("Warming up the neo4j database");
-                if (null == neo4jBean) {
-                    neo4jBean = beanFactory.createBean(Neo4jTestContainer.class);
-                    beanFactory.registerSingleton(Neo4jTestContainer.class.getName(), neo4jBean);
-                    // ((DefaultListableBeanFactory)beanFactory).registerDisposableBean(Neo4jTestContainer.class.getName(), neo4jBean);
+                EmbeddedNeo4j neo4jAnnotation = AnnotatedElementUtils.findMergedAnnotation(testClass, EmbeddedNeo4j.class);
+                if (null != neo4jAnnotation) {
+                    log.debug("detected the EmbeddedNeo4j annotation on class {}", testClass.getName());
+                    log.info("Warming up the neo4j database");
+                    if (null == neo4jBean) {
+                        neo4jBean = beanFactory.createBean(Neo4jTestContainer.class);
+                        beanFactory.registerSingleton(Neo4jTestContainer.class.getName(), neo4jBean);
+                        // ((DefaultListableBeanFactory)beanFactory).registerDisposableBean(Neo4jTestContainer.class.getName(), neo4jBean);
+                    }
+                    testValues =
+                        testValues.and(
+                            "spring.neo4j.uri=" + neo4jBean.getNeo4jContainer().getBoltUrl()
+                        );
                 }
-                testValues =
-                    testValues.and(
-                        "spring.neo4j.uri=" + neo4jBean.getNeo4jContainer().getBoltUrl()
-                    );
-            }
 <%_ } _%>
 <%_ if (messageBrokerKafka) { _%>
-            EmbeddedKafka kafkaAnnotation = AnnotatedElementUtils.findMergedAnnotation(testClass, EmbeddedKafka.class);
-            if (null != kafkaAnnotation) {
-                log.debug("detected the EmbeddedKafka annotation on class {}", testClass.getName());
-                log.info("Warming up the kafka broker");
-                if (null == kafkaBean) {
-                    kafkaBean = beanFactory.createBean(KafkaTestContainer.class);
-                    beanFactory.registerSingleton(KafkaTestContainer.class.getName(), kafkaBean);
-                    // ((DefaultListableBeanFactory)beanFactory).registerDisposableBean(KafkaTestContainer.class.getName(), kafkaBean);
+                EmbeddedKafka kafkaAnnotation = AnnotatedElementUtils.findMergedAnnotation(testClass, EmbeddedKafka.class);
+                if (null != kafkaAnnotation) {
+                    log.debug("detected the EmbeddedKafka annotation on class {}", testClass.getName());
+                    log.info("Warming up the kafka broker");
+                    if (null == kafkaBean) {
+                        kafkaBean = beanFactory.createBean(KafkaTestContainer.class);
+                        beanFactory.registerSingleton(KafkaTestContainer.class.getName(), kafkaBean);
+                        // ((DefaultListableBeanFactory)beanFactory).registerDisposableBean(KafkaTestContainer.class.getName(), kafkaBean);
+                    }
+                    testValues = testValues.and("spring.cloud.stream.kafka.binder.brokers=" + kafkaBean.getKafkaContainer().getHost() + ':' + kafkaBean.getKafkaContainer().getMappedPort(KafkaContainer.KAFKA_PORT));
                 }
-                testValues = testValues.and("spring.cloud.stream.kafka.binder.brokers=" + kafkaBean.getKafkaContainer().getHost() + ':' + kafkaBean.getKafkaContainer().getMappedPort(KafkaContainer.KAFKA_PORT));
-            }
 <%_ } _%>
 <%_ if (searchEngineElasticsearch) { _%>
-            EmbeddedElasticsearch elasticsearchAnnotation = AnnotatedElementUtils.findMergedAnnotation(testClass, EmbeddedElasticsearch.class);
-            if (null != elasticsearchAnnotation) {
-                log.debug("detected the EmbeddedElasticsearch annotation on class {}", testClass.getName());
-                log.info("Warming up the elastic database");
-                if (null == elasticsearchBean) {
-                    elasticsearchBean = beanFactory.createBean(ElasticsearchTestContainer.class);
-                    beanFactory.registerSingleton(ElasticsearchTestContainer.class.getName(), elasticsearchBean);
-                    // ((DefaultListableBeanFactory)beanFactory).registerDisposableBean(ElasticsearchTestContainer.class.getName(), elasticsearchBean);
+                EmbeddedElasticsearch elasticsearchAnnotation = AnnotatedElementUtils.findMergedAnnotation(testClass, EmbeddedElasticsearch.class);
+                if (null != elasticsearchAnnotation) {
+                    log.debug("detected the EmbeddedElasticsearch annotation on class {}", testClass.getName());
+                    log.info("Warming up the elastic database");
+                    if (null == elasticsearchBean) {
+                        elasticsearchBean = beanFactory.createBean(ElasticsearchTestContainer.class);
+                        beanFactory.registerSingleton(ElasticsearchTestContainer.class.getName(), elasticsearchBean);
+                        // ((DefaultListableBeanFactory)beanFactory).registerDisposableBean(ElasticsearchTestContainer.class.getName(), elasticsearchBean);
+                    }
+                    testValues =
+                        testValues.and(
+                            "spring.elasticsearch.uris=http://" + elasticsearchBean.getElasticsearchContainer().getHttpHostAddress()
+                        );
                 }
-                testValues =
-                    testValues.and(
-                        "spring.elasticsearch.uris=http://" + elasticsearchBean.getElasticsearchContainer().getHttpHostAddress()
-                    );
-            }
 <%_ } _%>
-            testValues.applyTo(context);
-        };
+                testValues.applyTo(context);
+            };
+        }
+        return contextCustomizer;
     }
 }

